src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,44 @@
+.calendar {
+  padding: 20px;
+
+  &__content {
+    display: flex;
+    flex-wrap: wrap;
+    gap: 1px;
+    width: $cell-size * 7 + 7;
+    margin: 0 auto;
+  }
+
+  &__day {
+    width: $cell-size;
+    height: $cell-size;
+    border: 1px solid black;
+    text-align: center;
+    line-height: $cell-size;
+    background-color: #eee;
+
+    transition: background-color, transform 0.5s, 0.5s;
+    cursor: pointer;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: "#{$i}";
+      }
+    }
+
+    &:hover {
+      transform: translateY(-20px);
+      background-color: #ffbccc;
+    }
+  }
+
+  @each $month-length in $month-lengths {
+    &--month-length--#{$month-length} {
+      @for $i from $month-length +1 through 31 {
```

Comment:
Usually you don't put a loop in the class definition, but on the contrary, you make a loop to define many styles for many classes.
That is, now you have a complex system that does not work. It needs to be divided and to think about what is wrong here.